### PR TITLE
Document Scarb.toml `publish` field

### LINKED ISSUE #1530: Document `publish` field in `Scarb.toml`
Document `publish` field described in #636 (implemented in #1421)

### PR DESCRIPTION
**Pull request by DelevoXDG**
Closes #1530

### CODE REVIEW

File: website/docs/reference/manifest.md [hunk 1]
```diff
@@ -70,6 +72,17 @@ edition = "{{ constants.edition }}"
 Most manifests have the edition field filled in automatically by `scarb new` with the latest available edition.
 If the edition field is not present in Scarb.toml, then the default edition is assumed.
 
+### `publish`
+
+The publish field is an optional key that determines whether the package can be packaged and published to a registry.
```
Comment (maciektr) [inline website/docs/reference/manifest.md:77]:
```suggestion
The publish field is an optional key that determines whether the package can be published to a registry.
```

File: website/docs/reference/manifest.md [hunk 2]
```diff
@@ -70,6 +72,17 @@ edition = "{{ constants.edition }}"
...
+Setting this field to false will prevent the package from being packaged or published.
```
Comment (maciektr) [inline website/docs/reference/manifest.md:78]:
```suggestion
Setting this field to false will prevent the package from being published.
```

**Review by maciektr — CHANGES_REQUESTED** [review]
Afaik packages with `publish = false` can ba packaged

File: website/docs/reference/manifest.md (hunk 1 above)
Comment (DelevoXDG) [reply website/docs/reference/manifest.md:77]: `publish = false` disallows `scarb package` as well, as it was outlined in #636

File: website/docs/reference/manifest.md (hunk 2 above)
Comment (DelevoXDG) [reply website/docs/reference/manifest.md:78]: https://github.com/software-mansion/scarb/pull/1533#discussion_r1727024853

File: website/docs/reference/manifest.md (hunk 1 above)
Comment (maciektr) [reply website/docs/reference/manifest.md:77]: https://github.com/software-mansion/scarb/blob/295935d87b9731ea05839f7901f735bebf3d7060/scarb/tests/package.rs#L1331

We discussed it after merging initial impl and I changed it

Comment (maciektr) [reply website/docs/reference/manifest.md:77]: https://github.com/software-mansion/scarb/pull/1447

Comment (DelevoXDG) [reply website/docs/reference/manifest.md]: Oh snap, totally forgot we changed this

**Review by maciektr — APPROVED** [review]

**Pull request merged**